controllers/operator/manager.go

```diff
@@ -429,6 +429,37 @@ func (m *ODLMOperator) GetClusterServiceVersionList(ctx context.Context, sub *ol
 	return csvs, nil
 }
 
+func (m *ODLMOperator) DeleteRedundantCSV(ctx context.Context, csvName, operatorNs, serviceNs, packageName string) error {
+	// Get the csv by its name and namespace
+	csv := &olmv1alpha1.ClusterServiceVersion{}
+	csvKey := types.NamespacedName{
+		Name:      csvName,
+		Namespace: serviceNs,
+	}
+	if err := m.Reader.Get(ctx, csvKey, csv); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(2).Infof("ClusterServiceVersion %s is not found", csvName)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get ClusterServiceVersion %s/%s", serviceNs, csvName)
+	}
+
+	if csv.GetLabels() == nil {
+		klog.V(2).Infof("ClusterServiceVersion %s in the namespace %s has no label", csvName, serviceNs)
+		return nil
+	}
+	// Delete the CSV if the csv does not contain label operators.coreos.com/packageName.operatorNs='' AND does not contains label olm.copiedFrom: operatorNs
+	if _, ok := csv.GetLabels()[fmt.Sprintf("operators.coreos.com/%s.%s", packageName, operatorNs)]; !ok {
```

Comment:
This code does not account for the label lenght limit of 64 characters, where for the long namespace names, the label name is truncated, see https://github.ibm.com/IBMPrivateCloud/roadmap/issues/61714